plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLQueryModelRecognizer.java

```diff
@@ -559,7 +572,7 @@ public void traverseObjs(Object obj) {
     }
 
     @Nullable
-    public SQLQuerySelectionModel recognizeQuery(@NotNull String text) {
+    public SQLQuerySelectionModel recognizeQuery(@NotNull String text, IProgressMonitor monitor) {
```

Comment:
I'd use `DBRProgressMonitor` everywhere as possible. When calling such a method, convert the monitor.

I also want to note we mostly pass monitors as the very first parameter of a method. Additionally, nullability annotations would be nice to have here.

This applies to most changes in this PR